ckanext/canada/plugins.py

```diff
@@ -963,3 +983,38 @@ def remove_outdated_resource_from_datastore(resource_id):
             log.info('Datastore table dropped for resource %s' % res['id'])
         except ObjectNotFound:
             log.error('Datastore table for resource %s does not exist' % res['id'])
+
+
+def remove_unsupported_resource_from_datastore(resource_id):
+    """
+    Double check the resource format. Only supported Xloader/Validation formats should have validation reports.
+    If the resource format is not supported, we should delete the validation reports and datastore tables.
+    """
+    lc = ckanapi.LocalCKAN()
+    try:
+        res = lc.action.resource_show(id=resource_id)
+        pkg = lc.action.package_show(id=res['package_id'])
+    except ObjectNotFound:
+        log.error('Resource %s does not exist.' % res['id'])
+        return
+
+    # only plain datasets will have validation applied to resources
+    # this avoids accidental deletion of PD datastore tables
+    if pkg['type'] != 'dataset':
+        return
+
+    if res.get(u'format', u'').lower() not in SUPPORTED_XLOADER_FORMATS:
+        log.info('Unsupported resource format "{}". Deleting validation reports and datastore tables for resource {}'
+                 .format(res.get(u'format', u'').lower(), res['id']))
+        try:
+            lc.action.resource_validation_delete(resource_id=res['id'])
+            log.info('Validation report deleted for resource %s' % res['id'])
+        except ObjectNotFound:
+            log.error('Validation report for resource %s does not exist' % res['id'])
+
+        if res['url_type'] == 'upload' or res['url_type'] == '':
```

Comment:
@wardi would it be prudent to add `and res['datastore_active']` to this condition?

Reply:
yes, but I still think this feature shouldn't be in the canada extension: it's much better implemented in xloader and ckanext-validation so that the community can help maintain it.

Reply:
@wardi Okay I will go see about putting it in the xloader extension. Do you think it should be in this same hook that fires before the database commit?

Reply:
That's your call, if there's a more obvious place to incorporate it into xloader then you could change it.

Oh, but for validation I guess removing a table does only make sense for our site (because we prevent xloader runs until validation passes). That could be a small override of the decision in xloader to schedule a load or table deletion just on our fork.

Reply:
Okay cool, that makes it a bit easier to implement upstream if we are only dealing with the Datastore tables there. Probably going to put it behind a config option like `ckanext.xloader.limit_datastore_tables`. Using `ckanext.xloader.formats` to check the supported formats for Xloader.

I can add a similar thing to the validation extension which would handle the removal of the validation reports based on the supported validation types (`ckanext.validation.formats`). And have another similar config option to control this like `ckanext.validation.limit_validation_reports`.